WordPress/Classes/ViewRelated/Post/PostSettingsViewController+Swift.swift

```diff
@@ -34,55 +38,77 @@ extension PostSettingsViewController {
         presentingViewController?.dismiss(animated: true)
     }
 
     @objc private func buttonSaveTapped() {
-        navigationItem.rightBarButtonItem = .activityIndicator
+        guard RemoteFeatureFlag.syncPublishing.enabled() else {
+            return _buttonSaveTapped()
+        }
+
+        guard isStandalone else {
+            saveChangesToParentContext()
+            presentingViewController?.dismiss(animated: true)
+            return
+        }
 
+        navigationItem.rightBarButtonItem = .activityIndicator
         setEnabled(false)
 
-        if RemoteFeatureFlag.syncPublishing.enabled() {
-            guard let snapshot else {
-                return assertionFailure("Snapshot missing")
-            }
-            Task { @MainActor in
-                do {
-                    let changes = RemotePostCreateParameters(post: apost).changes(from: snapshot)
-                    try await PostCoordinator.shared._update(apost, changes: changes)
-                    presentingViewController?.dismiss(animated: true)
-                } catch {
-                    setEnabled(true)
-                    refreshNavigationBarButtons()
-                }
-            }
-        } else {
-            PostCoordinator.shared.save(apost) { [weak self] result in
-                switch result {
-                case .success:
-                    self?.isStandaloneEditorDismissingAfterSave = true
-                    self?.presentingViewController?.dismiss(animated: true)
-                case .failure:
-                    self?.setEnabled(true)
-                    SVProgressHUD.showError(withStatus: Strings.errorMessage)
-                    self?.refreshNavigationBarButtons()
+        Task { @MainActor in
+            do {
+                let original = (snapshot.original ?? snapshot)
+                if original.status == .draft {
+                    saveChangesToParentContext()
+                    // TODO: Replace with new PostRepository._save
+                    PostCoordinator.shared.save(original)
+                } else {
+                    try await PostCoordinator.shared._update(original, changes: changes)
                 }
+                presentingViewController?.dismiss(animated: true)
+            } catch {
+                setEnabled(true)
+                refreshNavigationBarButtons()
+
             }
         }
     }
 
+    /// Returns a diff from the snapshot to the updated version of the post.
+    private var changes: RemotePostUpdateParameters {
+        RemotePostUpdateParameters.changes(from: snapshot, to: apost)
+    }
+
+    private func saveChangesToParentContext() {
+        do {
+            try self.apost.managedObjectContext?.save()
+        } catch {
+            // Should never happen
+            WordPressAppDelegate.crashLogging?.logError(error)
+        }
+    }
+
+    /// - note: Deprecated (kahu-offline-mode)
+    private func _buttonSaveTapped() {
+        saveChangesToParentContext()
```

Comment:
It's slightly different from the previous production behavior – it's now fully offline-first and it should be OK. It means that the "Save" is now also offline-first. I hope we don't even end up shipping this and be ready for 24.6 release.